test/e2e/gameserver_test.go

```diff
@@ -853,6 +849,32 @@ func TestGameServerPassthroughPort(t *testing.T) {
 	assert.Equal(t, "ACK: Hello World !\n", reply)
 }
 
+func TestGameServerPortPolicyNone(t *testing.T) {
+	t.Parallel()
+
+	runtime.FeatureTestMutex.Lock()
+	defer runtime.FeatureTestMutex.Unlock()
+	require.NoError(t, runtime.ParseFeatures(string(runtime.FeaturePortPolicyNone)+"=true"))
```

Comment:
For e2es, we don't flip feature gates this way. What we do instead is do one run with "all flipped" and another run with "all default". You can see this as `e2e-feature-gate` and `e2e-stable` in the "sub-build" for e2es. We do this because we do a single helm install for the whole e2e run, so we can't really flip feature gates dynamically like we do in unit tests.

Instead of this, you should just skip the test if the feature gate is not enabled. You can see an example of this here: https://github.com/googleforgames/agones/blob/87ba7d2c6b426c7389ba257000f72603bf37e012/test/e2e/gameserver_test.go#L1153-L1155